app.rb

```diff
@@ -1,15 +1,77 @@
+require 'json'
 require_relative 'person'
 require_relative 'classroom'
 require_relative 'book'
 require_relative 'rental'
 require_relative 'student'
 require_relative 'teacher'
 
+BOOKS_FILE = 'books.json'.freeze
+PEOPLE_FILE = 'people.json'.freeze
+RENTALS_FILE = 'rentals.json'.freeze
+
+def save_data(data, file_path)
+  File.write(file_path, JSON.generate(data))
+  puts "Data saved to #{file_path}!"
+end
+
+def load_data(file_path)
+  return [] unless File.exist?(file_path)
+
+  file_data = File.read(file_path)
+  json_data = JSON.parse(file_data)
+
+  case file_path
+  when BOOKS_FILE
+    load_books(json_data)
+  when PEOPLE_FILE
+    load_people(json_data)
+  when RENTALS_FILE
+    load_rentals(json_data)
+  end
+end
+
+def load_books(json_data)
+  json_data.map.with_index do |book_data, index|
+    Book.new(book_data['title'], book_data['author'], index + 1)
+  end
+end
+
+def load_people(json_data)
+  json_data.map.with_index do |person_data, index|
+    if person_data['classroom'].nil?
+      Teacher.new(person_data['age'], person_data['specialization'], name: person_data['name'], index: index + 1)
+    else
+      classroom = Classroom.new(person_data['classroom']['classroom&.label'])
+      Student.new(
+        person_data['age'],
+        classroom,
+        parent_permission: person_data['parent_permission'],
+        name: person_data['name'],
+        date: person_data['date'],
```

Comment:
- [ ] When there is data in the storage JSON files and I try to start the app it crashes. The app should work in the presence or absence of JSON files. You have handled the absence scenario well enough. Kindly do the same for when the data is present.

  ![image](https://github.com/Santosh-Konappanavar/school_library/assets/36542986/7fb600db-162e-42b1-964c-2fefa71114e3)